types/index.d.ts

```diff
@@ -230,8 +232,8 @@ declare module 'mongoose' {
     discriminator<T, U>(name: string | number, schema: Schema<T, U>, value?: string | number | ObjectId): U;
   }
 
-  type AnyKeys<T> = { [P in keyof T]?: T[P] | any };
   interface AnyObject { [k: string]: any }
+  type FlexibleObject<T extends {}> = { [P in keyof (T & Omit<any, keyof T>)]?: P extends keyof T ? T[P] : any }
 
```

Comment:
Maybe extends AnyObjects?

Reply:
I think no need to extends AnyObject, I have extends {} only to prevent generics like string, number and so on.

Reply:
But we might be able to merge both types by having   
````
type FlexibleObject<T extends {} = {}> =.....

````

Reply:
Then I need to refactor all AnyObject implementations. But this is not relevant to this feature title.